src/run.rs

```diff
@@ -30,6 +30,29 @@ fn edit<P: AsRef<OsStr>>(path: P) -> Result<(), i32> {
   }
 }
 
+pub(crate) const INIT_JUSTFILE: &str = "default:\n\techo 'Hello, world!'\n";
+
+pub fn init() -> Result<(), i32> {
+  let current_dir = &env::current_dir().unwrap();
+  let root = search::project_root(current_dir)
+    .map_err(|err| {
+      eprintln!("error searching for project root {:?}", err);
+      EXIT_FAILURE
+    })?;
+
+  search::search_for_justfile(root, |_| Err(SearchError::JustfileAlreadyExists)).map_err(|err| {
```

Comment:
I'm inclined not to add the `JustfileAlreadyExists` variant to the `SearchError`, since it implies that functions in search might return it, when for most of those functions it wouldn't be an error condition.

Reply:
I find the predicate here a bit confusing. I think I'd prefer to keep `search::justfile` as it was, and instead check for `SearchError::NotFound`.